datafusion/physical-plan/src/joins/join_hash_map.rs

```diff
@@ -261,13 +264,32 @@ pub trait JoinHashMapType {
         limit: usize,
         offset: JoinHashMapOffset,
     ) -> (Vec<u32>, Vec<u64>, Option<JoinHashMapOffset>) {
-        let mut input_indices = vec![];
-        let mut match_indices = vec![];
-
-        let mut remaining_output = limit;
+        let mut input_indices = Vec::with_capacity(limit);
+        let mut match_indices = Vec::with_capacity(limit);
```

Comment:
Might be worth investigating in a future PR if these vectors can be reused between lookups.

Reply:
Yeah sure - that's an interesting one (probably it can, but probably would need to get the buffer back from `PrimitiveArray` after passing it to `take`, i.e. via `get_mut`)